arm-network/2016-09-01/swagger/network.json

```diff
@@ -7798,10 +7805,16 @@
           "description": "Gets or sets resource guid property of the VirtualNetworkGateway resource"
         },
         "provisioningState": {
+          "readOnly": true,
           "type": "string",
           "description": "Gets provisioning state of the VirtualNetworkGateway resource Updating/Deleting/Failed"
         }
       },
+      "required": [
+        "ipConfigurations",
+        "gatewayType",
+        "vpnType"
+      ],
       "description": "VirtualNetworkGateway properties"
     },
     "VirtualNetworkGateway": {
```

Comment:
same reason.
"properties" property of "VirtualNetworkGateway" should be required.